spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,260 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordTypedStore) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecordTypedStore classes") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gather only TypedStore classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+
+        class CustomePost < Post
+        end
+        class Shop < ActiveRecord::Base
+        end
+
+        class User
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["CustomePost", "Post"]))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generates no definitions if there are no accessors to define") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+
+      RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with string type") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: true)
+          end
```

Comment:
Let's add a second `typed_store` to this class to ensure that we are processing all of them defined on each class.

Reply:
I think this double `typed_store` generating an output for all of the attributes defined from all of the typed store is important to test.